libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -132,76 +128,72 @@ bool NameAndTypeResolver::performImports(SourceUnit& _sourceUnit, map<string, So
 	return !error;
 }
 
-bool NameAndTypeResolver::resolveNamesAndTypes(ContractDefinition& _contract)
+bool NameAndTypeResolver::resolveNamesAndTypes(ASTNode& _node, bool _resolveInsideCode)
 {
+	bool success = true;
 	try
 	{
-		m_currentScope = m_scopes[_contract.scope()].get();
-		solAssert(!!m_currentScope, "");
+		if (ContractDefinition* contract = dynamic_cast<ContractDefinition*>(&_node))
+		{
+			m_currentScope = m_scopes[contract->scope()].get();
+			solAssert(!!m_currentScope, "");
 
-		ReferencesResolver resolver(m_errors, *this, nullptr);
-		bool success = true;
-		for (ASTPointer<InheritanceSpecifier> const& baseContract: _contract.baseContracts())
-			if (!resolver.resolve(*baseContract))
-				success = false;
+			for (ASTPointer<InheritanceSpecifier> const& baseContract: contract->baseContracts())
+				if (!resolveNamesAndTypes(*baseContract, false))
+					success = false;
 
-		m_currentScope = m_scopes[&_contract].get();
+			m_currentScope = m_scopes[contract].get();
 
-		if (success)
-		{
-			linearizeBaseContracts(_contract);
-			vector<ContractDefinition const*> properBases(
-				++_contract.annotation().linearizedBaseContracts.begin(),
-				_contract.annotation().linearizedBaseContracts.end()
-			);
-
-			for (ContractDefinition const* base: properBases)
-				importInheritedScope(*base);
-		}
+			if (success)
+			{
+				linearizeBaseContracts(*contract);
+				vector<ContractDefinition const*> properBases(
+					++contract->annotation().linearizedBaseContracts.begin(),
+					contract->annotation().linearizedBaseContracts.end()
+				);
 
-		// these can contain code, only resolve parameters for now
-		for (ASTPointer<ASTNode> const& node: _contract.subNodes())
-		{
-			m_currentScope = m_scopes[m_scopes.count(node.get()) ? node.get() : &_contract].get();
-			if (!resolver.resolve(*node))
-				success = false;
-		}
+				for (ContractDefinition const* base: properBases)
+					importInheritedScope(*base);
+			}
 
-		if (!success)
-			return false;
+			// these can contain code, only resolve parameters for now
+			for (ASTPointer<ASTNode> const& node: contract->subNodes())
+			{
+				m_currentScope = m_scopes[contract].get();
+				if (!resolveNamesAndTypes(*node, false))
+					success = false;
+			}
 
-		m_currentScope = m_scopes[&_contract].get();
+			if (!success)
+				return false;
 
-		// now resolve references inside the code
-		for (ModifierDefinition const* modifier: _contract.functionModifiers())
-		{
-			m_currentScope = m_scopes[modifier].get();
-			ReferencesResolver resolver(m_errors, *this, nullptr, true);
-			if (!resolver.resolve(*modifier))
-				success = false;
-		}
+			if (!_resolveInsideCode)
+				return success;
 
-		for (FunctionDefinition const* function: _contract.definedFunctions())
+			m_currentScope = m_scopes[contract].get();
+
+			// now resolve references inside the code
```

Comment:
The comment looks like talking about the new `_resolveInsideCode` parameter, but the comment exists before this PR.  I'm confused.